FindClangFormat.cmake

```diff
@@ -5,79 +5,78 @@
+#      message("clang-format executable found: ${ClangFormat_EXECUTABLE}\n"
+#              "version: ${ClangFormat_VERSION}")
 #    endif()
 
-find_program(CLANG_FORMAT_EXECUTABLE
+find_program(ClangFormat_EXECUTABLE
   NAMES clang-format clang-format-5.0
         clang-format-4.0 clang-format-3.9
         clang-format-3.8 clang-format-3.7
         clang-format-3.6 clang-format-3.5
         clang-format-3.4 clang-format-3.3
   DOC "clang-format executable"
 )
-mark_as_advanced(CLANG_FORMAT_EXECUTABLE)
+mark_as_advanced(ClangFormat_EXECUTABLE)
 
 # Extract version from command "clang-format -version"
-if(CLANG_FORMAT_EXECUTABLE)
-  execute_process(COMMAND ${CLANG_FORMAT_EXECUTABLE} -version
-                  OUTPUT_VARIABLE clang_format_version
+if(ClangFormat_EXECUTABLE)
+  execute_process(COMMAND ${ClangFormat_EXECUTABLE} -version
+                  OUTPUT_VARIABLE clangformat_version
                   ERROR_QUIET
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-  if (clang_format_version MATCHES "^clang-format version .*")
-    # clang_format_version sample: "clang-format version 3.9.1-4ubuntu3~16.04.1 (tags/RELEASE_391/rc2)"
+  if (clangformat_version MATCHES "^clang-format version .*")
+    # clangformat_version sample: "clang-format version 3.9.1-4ubuntu3~16.04.1 (tags/RELEASE_391/rc2)"
     string(REGEX REPLACE
            "clang-format version ([.0-9]+).*" "\\1"
-           CLANG_FORMAT_VERSION "${clang_format_version}")
-    # CLANG_FORMAT_VERSION sample: "3.9.1"
-
+           ClangFormat_VERSION "${clangformat_version}")
+    # ClangFormat_VERSION sample: "3.9.1"
     # Extract version components
-    string(REPLACE "." ";" clang_format_version "${CLANG_FORMAT_VERSION}")
-    list(LENGTH clang_format_version CLANG_FORMAT_VERSION_COUNT)
-    if(CLANG_FORMAT_VERSION_COUNT GREATER 0)
-      list(GET clang_format_version 0 CLANG_FORMAT_VERSION_MAJOR)
+    string(REPLACE "." ";" clangformat_version "${ClangFormat_VERSION}")
+    list(LENGTH clangformat_version ClangFormat_VERSION_COUNT)
+    if(ClangFormat_VERSION_COUNT GREATER 0)
+      list(GET clangformat_version 0 ClangFormat_VERSION_MAJOR)
     else()
-      set(CLANG_FORMAT_VERSION_MAJOR 0)
+      set(ClangFormat_VERSION_MAJOR 0)
     endif()
-    if(CLANG_FORMAT_VERSION_COUNT GREATER 1)
-      list(GET clang_format_version 1 CLANG_FORMAT_VERSION_MINOR)
+    if(ClangFormat_VERSION_COUNT GREATER 1)
+      list(GET clangformat_version 1 ClangFormat_VERSION_MINOR)
     else()
-      set(CLANG_FORMAT_VERSION_MINOR 0)
+      set(ClangFormat_VERSION_MINOR 0)
     endif()
-    if(CLANG_FORMAT_VERSION_COUNT GREATER 2)
-      list(GET clang_format_version 2 CLANG_FORMAT_VERSION_PATCH)
+    if(ClangFormat_VERSION_COUNT GREATER 2)
+      list(GET clangformat_version 2 ClangFormat_VERSION_PATCH)
     else()
-      set(CLANG_FORMAT_VERSION_PATCH 0)
+      set(ClangFormat_VERSION_PATCH 0)
     endif()
   endif()
-  unset(clang_format_version)
+  unset(clangformat_version)
```

Comment:
Shouldn't we also unset `ClangFormat_VERSION_COUNT`? Or does that not influence the `ClangFormat_VERSION` variable used by `find_package_handle_standard_args`?

Reply:
`find_package_handle_standard_args` takes `ClangFormat_VERSION` as the `FOUND_VERSION`, and it does not use `ClangFormat_VERSION_COUNT`. <package>_VERSION_COUNT is set in find_package with CONFIG MODE